src/mono/mono/mini/interp/interp.c

```diff
@@ -4259,7 +4195,45 @@ mono_interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClause
 #else
 			ip += 4;
 #endif
-call:
+
```

Comment:
The common call path (for `MINT_CALL`) shouldn't execute this code. When we do a static call we know whether the method to be called is aot or not so we are just adding unnecessary checks to a hot path.

Reply:
Moved it away from `MINT_CALL`.